openmc/nuclide.py

```diff
@@ -98,7 +99,14 @@ def scattering(self):
     @name.setter
     def name(self, name):
         check_type('name', name, basestring)
-        self._name = name
+
+        if '-' in name:
+            new_name = name.replace('-', '')
```

Comment:
To be complete, I'd change this to:
```Python
new_name = name.replace('-', '')
new_name = new_name.replace('Nat', '0')
if new_name.endswith('m'):
    new_name = new_name[:-1] + '_m1'
```
Also will need to change the self._name assignment below.